File: Circular Sentence.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        words = sentence.split()
        n = len(words)
        for i in range(1, n):
            if words[i][0] != words[i-1][-1]:
                return False
        return words[-1][-1] == words[0][0]

File: Circular Sentence.go
func isCircularSentence(sentence string) bool {
    n := len(sentence)
    for i := 0; i < n; i++ {
        if sentence[i] == ' ' {
            if sentence[i-1] != sentence[i+1] {
                return false
            }
        }
    }

    return sentence[0] == sentence[n-1]
}
